rclcpp/include/rclcpp/executor.hpp

```diff
@@ -319,6 +320,50 @@ class Executor
   virtual void
   spin_once(std::chrono::nanoseconds timeout = std::chrono::nanoseconds(-1));
 
+  /// Spin (blocking) until the condition is complete, it times out waiting,
+  /// or rclcpp is interrupted.
+  /**
+   * \param[in] future The condition which can be callable or future type to wait on.
+   *   If this function returns SUCCESS, the future can be
+   *   accessed without blocking (though it may still throw an exception).
+   * \param[in] timeout Optional timeout parameter, which gets passed to Executor::spin_node_once.
+   *   `-1` is block forever, `0` is non-blocking.
+   *   If the time spent inside the blocking loop exceeds this timeout, return a TIMEOUT return
+   *   code.
+   * \return The return code, one of `SUCCESS`, `INTERRUPTED`, or `TIMEOUT`.
+   */
+  template<typename Condition, typename DurationT = std::chrono::milliseconds>
```

Comment:
nitpick: use `ConditionT`

We've been somewhat inconsistent on this, but the trailing `T` is preferable as it makes it easier to search through the project, especially if in the future we have an actual class called `Condition` (which is very possible).